backend/src/main/java/de/njsm/versusvirus/backend/service/purchase/PurchaseService.java

```diff
@@ -159,6 +156,15 @@ public void withdrawPurchase(UUID purchaseId) {
                 purchase.getVolunteerApplications().clear();
             }
             purchase.setStatus(NEW);
+        } else if (purchase.getStatus() == VOLUNTEER_ACCEPTED) {
+            telegramApi.deleteMessage(groupChatId, purchase.getBroadcastMessageId());
```

Comment:
This is unsafe. There might be submitted receipts for this purchase, potentially screwing with business logic further in the purchase process.

Reply:
True, you are correct. I forgot that we are still in this state after the receipts come in but before all receipts have been received. I'll correct that

Reply:
@ntruessel Fixed, please have a look :) https://github.com/ElsaHilftDigital/main/pull/127/commits/67d95fca30c7e80ece3129bbcaff04f993681150